lib/netsnmp/varbind.rb

```diff
@@ -81,11 +81,11 @@ def convert_to_asn(typ, value)
                    when :ipaddress then 0
                    when :counter32
                      asn_val = [value].pack("N*")
-                     asn_val = asn_val[1..-1] while asn_val.start_with?("\x00")
+                     asn_val = asn_val[1..-1] while asn_val.start_with?("\x00") && asn_val[1] == "\x00"
                      1
                    when :gauge
-                     asn_val = [value].pack("N*")
```

Comment:
I'm not 100% sure about this change. It's correct in that it uses 32 bit signed integer format, but it uses native endianess. There's no format to specify network endianness.